File: backend/src/main/java/com/totp/auth/dto/LoginResponse.java
package com.totp.auth.dto;

public record LoginResponse(
        String status,
        String message,
        boolean totpRequired,
        String accessToken,
        String refreshToken,
        String tempAuthSessionId,
        Long tempAuthSessionExpiresInSeconds,
        Long expiresIn
) {

    public LoginResponse(
            String status,
            String message,
            boolean totpRequired,
            String accessToken,
            String refreshToken
    ) {
        this(
                status,
                message,
                totpRequired,
                accessToken,
                refreshToken,
                null,
                null,
                null
        );
    }
}

File: mobile/lib/models/login_response.dart
class LoginResponse {
  final String status;
  final String message;
  final bool totpRequired;
  final String? accessToken;
  final String? refreshToken;
  final String? tempAuthSessionId;
  final int? tempAuthSessionExpiresInSeconds;

  const LoginResponse({
    required this.status,
    required this.message,
    required this.totpRequired,
    this.accessToken,
    this.refreshToken,
    this.tempAuthSessionId,
    this.tempAuthSessionExpiresInSeconds,
  });

  factory LoginResponse.fromJson(
    Map<String, dynamic> json,
  ) {
    return LoginResponse(
      status: json['status'] as String,
      message: json['message'] as String,
      totpRequired:
          json['totpRequired'] as bool,
      accessToken:
          json['accessToken'] as String?,
      refreshToken:
          json['refreshToken'] as String?,
      tempAuthSessionId:
          (json['tempAuthSessionId'] ??
                  json['temporaryAuthSessionToken'])
              as String?,
      tempAuthSessionExpiresInSeconds:
          (json['tempAuthSessionExpiresInSeconds'] ??
                  json['temporaryAuthSessionExpiresInSeconds'])
              as int?,
    );
  }
}